File: src/main/java/com/budilov/lambda/ses/EmailService.java
package com.budilov.lambda.ses;

import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.regions.Region;
import com.amazonaws.regions.Regions;
import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClient;
import com.amazonaws.services.simpleemail.model.*;

/**
 * Created by Vladimir Budilov on 10/25/15.
 */
public class EmailService {

    private static String TO = PropUtil.get("toEmail"); // Replace with a "To" address. If your account is still in the

    private static BasicAWSCredentials awsCreds = new BasicAWSCredentials(PropUtil.get("accessKey"), PropUtil.get("secretKey"));

    public static void sendEmail(Email email) throws Exception {

        Destination destination = new Destination().withToAddresses(new String[]{TO});

        Content subject = new Content().withData(email.getSubject() + " (from " + email.getFrom() + ")");
        Content textBody = new Content().withData(email.getMessage());
        Body body = new Body().withText(textBody);

        Message message = new Message().withSubject(subject).withBody(body);

        SendEmailRequest request = new SendEmailRequest().withSource(PropUtil.get("fromEmail"))
                .withDestination(destination)
                .withMessage(message);

        try {
            System.out.println("Attempting to send an email through Amazon SES by using the AWS SDK for Java...");
            AmazonSimpleEmailServiceClient client = new AmazonSimpleEmailServiceClient(awsCreds);
            Region REGION = Region.getRegion(Regions.US_EAST_1);
            client.setRegion(REGION);

            client.sendEmail(request);
            System.out.println("Email sent!");
        } catch (Exception ex) {
            System.out.println("The email was not sent.");
            System.out.println("Error message: " + ex.getMessage());
            throw ex;
        }
    }
}

File: src/main/java/com/budilov/lambda/ses/EmailService.kt
package com.budilov.lambda.ses

import com.amazonaws.regions.Region
import com.amazonaws.regions.Regions
import com.amazonaws.services.lambda.runtime.LambdaLogger
import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClient
import com.amazonaws.services.simpleemail.model.*

data class Email(var fromEmail: String = "", var subject: String = "", var message: String = "")

/**
 * Created by Vladimir Budilov on 10/25/15.
 */
class EmailService(val logger: LambdaLogger?) {

    @Throws(Exception::class)
    fun sendEmail(email: Email) {
        logger?.log("Sending the email: $email")
        val subject = Content().withData("""Subject: ${email.subject} (submitter: ${email.fromEmail})""")
        val textBody = Content().withData(getEmailBody(email.fromEmail, email.message))
        val body = Body().withText(textBody)

        // create the actual message
        val message = Message().withSubject(subject).withBody(body)

        val request = SendEmailRequest()
                .withSource(Properties.destinationEmail)
                .withDestination(Destination()
                        .withToAddresses(Properties.destinationEmail))
                .withMessage(message)
        try {
            logger?.log("Attempting to send the email")
            val client = AmazonSimpleEmailServiceClient()

            // Get the AWS_REGION environment variable
            client.setRegion(Region.getRegion(Regions.fromName(Properties.region)))
            client.sendEmail(request)
            logger?.log("message: Email sent!")
        } catch (ex: Exception) {
            logger?.log("The email was not sent: ${ex.message}")
            throw ex
        }

    }

    fun getEmailBody(email: String, message: String): String {
        return """

        email (submitter): $email

        message: $message
        """
    }
}
